src/Computation/PooledContext.cs

```diff
@@ -64,9 +64,11 @@ public BigInteger GetSingleValue(BitVecExpr variable)
             : throw new IrreducibleSymbolicExpressionException();
     }
 
-    public BigInteger GetExampleValue(BitVecExpr variable)
+    public BigInteger GetExampleValue(IValue variable)
     {
-        return ((BitVecNum) CreateModel().Eval(variable, true)).BigInteger;
+        var expr = variable.AsBitVector(this);
+
+        return ((BitVecNum) CreateModel().Eval(expr, true)).BigInteger;
```

Comment:
This was what I was always trying to avoid. If `expr` were to be inlined this becomes wrong since any implicit symbol assertions it makes are not applied to the model. I think my cure of enforcing that the method take the `Expr` was worse than the disease but I still don't like this. We are about to add a `using` so inlining does become less likely, but still seems dodgy.

Reply:
Impérative languages for you 🤷‍♂️ At least we've localised to this method. I do think a comment here to our future selves would be a good idea though.